Modules/Filtering/ImageStatistics/test/itkLabelOverlapMeasuresImageFilterGTest.cxx

```diff
@@ -0,0 +1,207 @@
+  EXPECT_NEAR(filter->GetMeanOverlap(), 2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetDiceCoefficient(), 2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetVolumeSimilarity(), -2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetFalseNegativeError(), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetFalsePositiveError(), 0, 0.0);
+  EXPECT_NEAR(filter->GetFalseDiscoveryRate(), 0, 0.0);
+
+
+  EXPECT_NEAR(filter->GetTargetOverlap(1), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetUnionOverlap(1), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetMeanOverlap(1), 2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetDiceCoefficient(1), 2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetVolumeSimilarity(1), -2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetFalseNegativeError(1), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetFalsePositiveError(1), 0, 0.0);
+  EXPECT_NEAR(filter->GetFalseDiscoveryRate(1), 0, 0.0);
+}
+
+
+TEST_F(LabelOverlapMeasuresImageFilterFixture, test3)
+{
+  auto source = Utils::CreateImage(0);
+  auto target = Utils::CreateImage(0);
+
+  auto idx1 = itk::MakeFilled<Utils::IndexType>(1);
+  source->SetPixel(idx1, 1);
+  target->SetPixel(idx1, 1);
+  ++idx1[0];
+  target->SetPixel(idx1, 1);
+  auto idx2 = itk::MakeFilled<Utils::IndexType>(3);
+  source->SetPixel(idx2, 2);
+  target->SetPixel(idx2, 2);
+  ++idx2[2];
+  source->SetPixel(idx2, 2);
+  target->SetPixel(idx2, 2);
+  ++idx2[2];
+  source->SetPixel(idx2, 2);
+  ++idx2[2];
+
+
+  auto filter = Utils::FilterType::New();
+  filter->SetSourceImage(source);
+  filter->SetTargetImage(target);
+  filter->Update();
+
+  EXPECT_NEAR(filter->GetTotalOverlap(), 0.75, 1e-15);
+  EXPECT_NEAR(filter->GetUnionOverlap(), 3.0 / 5.0, 1e-15);
+  EXPECT_NEAR(filter->GetMeanOverlap(), 0.75, 1e-15);
+  EXPECT_NEAR(filter->GetDiceCoefficient(), 0.75, 1e-15);
+  EXPECT_NEAR(filter->GetVolumeSimilarity(), 0, 1e-15);
+  EXPECT_NEAR(filter->GetFalseNegativeError(), 0.25, 1e-15);
+  EXPECT_NEAR(filter->GetFalsePositiveError(), 2.38418806475e-07, 1e-17);
+  EXPECT_NEAR(filter->GetFalseDiscoveryRate(), 0.25, 1e-15);
+
+
+  EXPECT_NEAR(filter->GetTargetOverlap(1), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetUnionOverlap(1), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetMeanOverlap(1), 2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetDiceCoefficient(1), 2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetVolumeSimilarity(1), -2.0 / 3.0, 1e-15);
+  EXPECT_NEAR(filter->GetFalseNegativeError(1), 0.5, 0.0);
+  EXPECT_NEAR(filter->GetFalsePositiveError(1), 0, 0.0);
+  EXPECT_NEAR(filter->GetFalseDiscoveryRate(1), 0, 0.0);
+
+  EXPECT_NEAR(filter->GetTargetOverlap(2), 1.0, 0.0);
+  EXPECT_NEAR(filter->GetUnionOverlap(2), 2.0 / 3.0, 2e-15);
+  EXPECT_NEAR(filter->GetMeanOverlap(2), 4.0 / 5.0, 1e-15);
+  EXPECT_NEAR(filter->GetDiceCoefficient(2), 4.0 / 5.0, 1e-15);
+  EXPECT_NEAR(filter->GetVolumeSimilarity(2), 2.0 / 5.0, 1e-15);
+  EXPECT_NEAR(filter->GetFalseNegativeError(2), 0.0, 0.0);
+  EXPECT_NEAR(filter->GetFalsePositiveError(2), 4.76837612950e-07, 1e-17);
```

Comment:
@dzenanz Are you sure this is computed correctly? 
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/ImageStatistics/include/itkLabelOverlapMeasuresImageFilter.hxx#L378

Reply:
Yes. All the image pixels, by definition, are the union of those 4 classes.

Reply:
I don't know about `EXPECT_NEAR(filter->GetFalsePositiveError(2), 4.76837612950e-07, 1e-17);`. Shouldn't it be a round number? @Joeycho

Reply:
@dzenanz If I estimated correctly, the only possible case with label 2 is
 TP: 2, FN: 0, FP:1, TN: almost infinitiv.

Is there anyway, to print it out nVox? https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Filtering/ImageStatistics/include/itkLabelOverlapMeasuresImageFilter.hxx#L378

It should be big enough.

Reply:
So, for label 2, TP:2, FN:0, FP:1, TN: 2097149

nVox should be 2097152.

Reply:
So the FalsePositiveRate is defined as: `FP/TN+FP` -> `1/(2097149+1)`= `4.768376129509096e-07` so that matched.

Reply:
> So the FalsePositiveRate is defined as: `FP/TN+FP` -> `1/(2097149+1)`= `4.768376129509096e-07` so that matched.

Yes, exactly! if TN is really 2097149, the number is valid. We should not round the number.

Reply:
The image is 128^3 size which is 2097152 pixels matching the total sum.